### PR TITLE
Backend support for blackboard user facing configurations

### PR DESCRIPTION
**Pull request by alex-sheehan-edx**
Thanks to @taliaedX for the initial work that I just easily modeled after.

WIP- 
TODO:
Unit testing
Tweaking based on frontend work
General cleanup

**Merge checklist:**
- [ ] Any new requirements are in the right place (do **not** manually modify the `requirements/*.txt` files)
    - `base.in` if needed in production but edx-platform doesn't install it
    - `test-master.in` if edx-platform pins it, with a matching version
    - `make upgrade && make requirements` have been run to regenerate requirements
- [ ] `make static` has been run to update webpack bundling if any static content was updated
- [ ] `./manage.py makemigrations` has been run
    - Checkout the [Database Migration](https://openedx.atlassian.net/wiki/spaces/AC/pages/23003228/Everything+About+Database+Migrations) Confluence page for helpful tips on creating migrations.
    - *Note*: This **must** be run if you modified any models.
      - It may or may not make a migration depending on exactly what you modified, but it should still be run.
    - This should be run from either a venv with all the lms/edx-enterprise requirements installed or if you checked out edx-enterprise into the src directory used by lms, you can run this command through an lms shell.
        - It would be `./manage.py lms makemigrations` in the shell.
- [ ] [Version](https://github.com/edx/edx-enterprise/blob/master/enterprise/__init__.py) bumped
- [ ] [Changelog](https://github.com/edx/edx-enterprise/blob/master/CHANGELOG.rst) record added
- [ ] Translations updated (see docs/internationalization.rst but also this isn't blocking for merge atm)

**Post merge:**
- [ ] Tag pushed and a new [version](https://github.com/edx/edx-enterprise/releases) released
    - *Note*: Assets will be added automatically. You just need to provide a tag (should match your version number) and title and description.
- [ ] After versioned build finishes in [Travis](https://travis-ci.org/github/edx/edx-enterprise), verify version has been pushed to [PyPI](https://pypi.org/project/edx-enterprise/)
    - Each step in the release build has a condition flag that checks if the rest of the steps are done and if so will deploy to PyPi.
    (so basically once your build finishes, after maybe a minute you should see the new version in PyPi automatically (on refresh))
- [ ] PR created in [edx-platform](https://github.com/edx/edx-platform) to upgrade dependencies (including edx-enterprise)
    - This **must** be done after the version is visible in PyPi as `make upgrade` in edx-platform will look for the latest version in PyPi.
    - Note: the edx-enterprise constraint in edx-platform **must** also be bumped to the latest version in PyPi.

### CODE REVIEW

File: integrated_channels/api/v1/blackboard/urls.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
+"""
+    url mappings for integrated_channels/v1/moodle/
```
Comment (taliaedX) [inline integrated_channels/api/v1/blackboard/urls.py:2]: "general cleanup" probably includes this reference, but just in case!

**Review by taliaedX — APPROVED** [review]
Considering I literally just did the same thing, but not blackboard, yeah I recognize everything here and it looks correct.

Just fix the doc strings. ;) 

Oh and we need to be careful of inevitable clobbering of each other with the base integrated_channels/urls.py file since we're all going to conflict.

File: integrated_channels/api/v1/blackboard/urls.py (hunk 1 above)
Comment (alex-sheehan-edx) [reply integrated_channels/api/v1/blackboard/urls.py:2]: 😡 I was so sure I got them all. Hahaha yes general cleanup did refer to things like this

**Comment by alex-sheehan-edx** [comment]
> Oh and we need to be careful of inevitable clobbering of each other with the base integrated_channels/urls.py file since we're all going to conflict.

I can wait for you to merge and then take care of conflicts, not too difficult 🙂

File: tests/test_integrated_channels/test_api/test_blackboard/test_views.py [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+                         self.enterprise_customer_conf.enterprise_customer.uuid)
+
+    @mock.patch('enterprise.rules.crum.get_current_request')
+    def test_post(self, mock_current_request):
+        self.enterprise_customer_conf.delete()
+        mock_current_request.return_value = self.get_request_with_jwt_cookie(
+            system_wide_role=ENTERPRISE_ADMIN_ROLE,
+        )
+        url = reverse('api:v1:blackboard:configuration-list')
+        payload = {
```
Comment (alex-sheehan-edx) [inline tests/test_integrated_channels/test_api/test_blackboard/test_views.py:77]: Looking at the other tests, I noticed that we're not requiring a base url here. That seems fairly important for an integration, is there any desire to make it a required param?

File: tests/test_integrated_channels/test_api/test_canvas/test_views.py [hunk 3]
```diff
@@ -74,6 +74,8 @@ def test_post(self, mock_current_request):
             'enterprise_customer': self.enterprise_customer.uuid,
         }
         response = self.client.post(url, payload, format='json')
+        import pdb;
```
Comment (alex-sheehan-edx) [inline tests/test_integrated_channels/test_api/test_canvas/test_views.py:77]: whoopsies 😬

File: tests/test_integrated_channels/test_api/test_blackboard/test_views.py (hunk 2 above)
Comment (taliaedX) [reply tests/test_integrated_channels/test_api/test_blackboard/test_views.py:77]: If you don't include it, it should already throw an error no?
And the front end should already prevent form submission without it.

Comment (jlajoie) [reply tests/test_integrated_channels/test_api/test_blackboard/test_views.py:77]: Bare minimum require it on the frontend I'd say, backend up to you, could potentially have people want to create their object and come back and flush it out later before enabling? No strong opinions here

**Review by jlajoie — APPROVED** [review]
👍 lgtm

File: tests/test_integrated_channels/test_api/test_blackboard/test_views.py (hunk 2 above)
Comment (alex-sheehan-edx) [reply tests/test_integrated_channels/test_api/test_blackboard/test_views.py:77]: ok, decided noop. Thanks!

**Comment by alex-sheehan-edx** [comment]
jenkins run django22

**Comment by alex-sheehan-edx** [comment]
jenkins run all

**Pull request merged**